server/src/test/java/com/objectcomputing/checkins/services/checkinnotes/CheckinNoteControllerTest.java

```diff
@@ -0,0 +1,272 @@
+
+    }
+
+    @Test
+    void testFindCheckinNote() {
+        CheckinNote cNote = new CheckinNote(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(),"test");
+        Set<CheckinNote> checkinNote = Collections.singleton(cNote);
+
+        when(checkinNoteServices.findByFields(cNote.getCheckinid(), cNote.getCreatedbyid())).thenReturn(checkinNote);
+
+        final HttpRequest<?> request = HttpRequest.GET(String.format("/?checkinid=%s&createdbyid=%s",cNote.getCheckinid(),cNote.getCreatedbyid()));
+        final HttpResponse<Set<CheckinNote>> response = client.toBlocking().exchange(request, Argument.setOf(CheckinNote.class));
+
+        assertEquals(checkinNote, response.body());
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+    }
+
+    @Test
+    void testFindCheckinNoteNull() {
+        CheckinNote cNote = new CheckinNote(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(),"test");
+        when(checkinNoteServices.findByFields(eq(cNote.getCheckinid()),eq(null))).thenReturn(null);
+        
+        final HttpRequest<?> request = HttpRequest.GET(String.format("/?checkinid=%s",cNote.getCheckinid()));
+        HttpClientResponseException responseException = assertThrows(HttpClientResponseException.class, () -> client.toBlocking().exchange(request, Argument.setOf(CheckinNote.class)));
+
+        assertEquals(HttpStatus.NOT_FOUND, responseException.getStatus());
+
+    }
+
+    @Test
+    void testUpdateCheckinNote(){
+    CheckinNote cNote = new CheckinNote(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(),"test");
+
+    when(checkinNoteServices.update(eq(cNote))).thenReturn(cNote);
+  
+    final HttpRequest<?> request = HttpRequest.PUT("",cNote);
+    final HttpResponse<CheckinNote> response = client.toBlocking().exchange(request, CheckinNote.class);
+
+    assertEquals(cNote, response.body());
+    assertEquals(HttpStatus.OK, response.getStatus());
+    }
+    
+    @Test
+    void testUpdateInvalidCheckinNote(){
+        CheckinNote cNote = new CheckinNote(UUID.randomUUID(), null, null,"test");
+        when(checkinNoteServices.update(any(CheckinNote.class))).thenReturn(cNote);
+        
+        final HttpRequest<?> request = HttpRequest.PUT("",cNote);
+        HttpClientResponseException responseException = assertThrows(HttpClientResponseException.class,
+                () -> client.toBlocking().exchange(request, Map.class));
+
+                
+        JsonNode body = responseException.getResponse().getBody(JsonNode.class).orElse(null);
+        JsonNode errors = Objects.requireNonNull(body).get("_embedded").get("errors");
+        JsonNode href = Objects.requireNonNull(body).get("_links").get("self").get("href");
+        List<String> errorList = List.of(errors.get(0).get("message").asText(), errors.get(1).get("message").asText())
+                        .stream().sorted().collect(Collectors.toList());
+        assertEquals("CheckinNote.checkinid: must not be null", errorList.get(0));
```

Comment:
This will be `checkinNote.checkinid: must not be null` 
Same with next line un-capitalize the C in CheckinNote

This test case will fail.